app/views/layouts/_form-error-summary.haml

```diff
@@ -4,4 +4,4 @@
   %ul.error-summary-list
     - flash.each do |field, message|
       %li
-      = link_to message, "##{field}"
\ No newline at end of file
+      = link_to message, "##{field.dasherize}"
```

Comment:
All this wrangling between `dasherize` and `underscore` is to convert from register fields with dashes e.g. `end-date` to something safe for a ruby symbol `end_date`